docs/GettingStarted.md

```diff
@@ -1725,6 +1727,50 @@ Datadog.tracer.trace('my.operation') { logger.warn('This is a traced operation.'
 # [2019-01-16 18:38:41 +0000][my_app][WARN][dd.trace_id=8545847825299552251 dd.span_id=3711755234730770098] This is a traced operation.
 ```
 
+### Metrics
+
+The tracer and its integrations can produce some additional metrics that can provide useful insight into the performance of your application. These metrics are collected with `dogstatsd-ruby`, and can be sent to the same Datadog agent to which you send your traces.
+
+To configure your application for metrics collection:
+
+1. [Configure your Datadog agent for StatsD](https://docs.datadoghq.com/developers/dogstatsd/#setup)
+2. Add `gem 'dogstatsd-ruby'` to your Gemfile
+
+#### For application runtime
+
+If runtime metrics are configured, the trace library will automatically collect and send metrics about the health of your application.
+
+To configure runtime metrics, add the following configuration:
+
+```ruby
+# config/initializers/datadog.rb
+require 'datadog/statsd'
+require 'ddtrace'
+
+Datadog.configure do |c|
+  # Statsd is automatically configured with default settings if `dogstatsd-ruby` is available.
+  # Otherwise, you can configure with host and port of Datadog agent; defaults to 'localhost:8125'.
+  c.runtime_metrics statsd: Datadog::Statsd.new
+end
+```
+
+See the [Dogstatsd documentation](https://www.rubydoc.info/github/DataDog/dogstatsd-ruby/master/frames) for more details about configuring `Datadog::Statsd`.
+
+The stats sent will include:
+
+| Name                        | Type    | Description                                              |
+| --------------------------  | ------- | -------------------------------------------------------- |
+| `runtime.ruby.class_count`  | `gauge` | Number of classes in memory space.                       |
+| `runtime.ruby.thread_count` | `gauge` | Number of threads.                                       |
+| `runtime.ruby.gc.*`.        | `gauge` | Garbage collection statistics (one per value in GC.stat) |
```

Comment:
Based on the chat with @rochdev 
It looks like some of Ruby's GC's metrics would be better represented by a a `count` aka `increment` type.

From `GC.stats` output it looks like these metrics should be represented as gauges:
```ruby
{
:count=>17, 
:minor_gc_count=>14,
:major_gc_count=>3
}
```

Other - seem to be represented well enough as gauges.

Reply:
What's the advantage of `count` over `gauge` here?

Reply:
Actually it's a `rate` and not a `count`. It will automatically normalize the metric for a specific time interval. The default is per second. So for example, if every 10 seconds you send 100 GCs, then the graph will show 10. With gauge you have to manually divide by 10 and some query features are not available I think too. Prometheus use `rate` (they call it counters) for counts and sums, and I think it makes sense.

Reply:
Forgive me if this is a naive question, but why does the concept of rate matter here? Don't we just want the snapshot of that value in that point of time? That's what `gauge` does, which makes rate calculations irrelevant.

Reply:
The problem is that there must be a known interval for the user to make sense of a metric. If there was 100 GCs, that happened over what amount of time? We could always include it in the name like `GCs per 10 seconds` but that's a weird interval as users would generally expect per second or per minute. It makes even less sense if the interval is changed or configurable for example. Then it makes it nearly impossible to know the actual interval. Using a rate allows configuring this at query time, and the interval is calculated from the time the metrics were sent.

For a better explanation, see https://prometheus.io/docs/concepts/metric_types/ and https://prometheus.io/docs/practices/histograms/ (counter is the equivalent of rate)

Reply:
> The problem is that there must be a known interval for the user to make sense of a metric.

I'm actually not sure I understand this or agree with it. A gauge is submitted at time A with 30, at time B with 35; both metrics appear on the graph as data points. What else do you need? I don't understand why the amount of time is relevant.

Reply:
Let's test your theory with a scenario:

You spent a total of 220ms in garbage collection. I don't tell you the time interval. Do you have an issue or not?

Reply:
As per offline discussion, it seems this use case is not needed for Ruby, and the change would be too complicated to be warranted. We'll revisit if a user explicitly asks for it.